reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/SdkCoreContextTest.kt

```diff
@@ -0,0 +1,393 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.integration
+
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.datadog.android.Datadog
+import com.datadog.android.api.SdkCore
+import com.datadog.android.api.context.UserInfo
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.api.feature.StorageBackedFeature
+import com.datadog.android.api.feature.stub.StubStorageBackedFeature
+import com.datadog.android.core.configuration.Configuration
+import com.datadog.android.core.integration.tests.MockServerTest
+import com.datadog.android.core.integration.tests.forge.factories.ConfigurationCoreForgeryFactory
+import com.datadog.android.privacy.TrackingConsent
+import com.datadog.tools.unit.forge.exhaustiveAttributes
+import com.datadog.tools.unit.forge.useToolsFactories
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.annotation.StringForgeryType
+import fr.xgouchet.elmyr.junit4.ForgeRule
+import fr.xgouchet.elmyr.jvm.useJvmFactories
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+/**
+ * Provides the tests for the SDK Core API responsible for modifying the User information, data collection consent,
+ * etc.
+ */
+@RunWith(AndroidJUnit4::class)
+class SdkCoreContextTest : MockServerTest() {
+
+    @get:Rule
+    var forge = ForgeRule().useJvmFactories().useToolsFactories().withFactory(ConfigurationCoreForgeryFactory())
+
+    @StringForgery(type = StringForgeryType.ALPHABETICAL)
+    lateinit var fakeUserId: String
+
+    @StringForgery(regex = "[A-Z][a-z]+ [A-Z]\\. [A-Z][a-z]+")
+    lateinit var fakeUserName: String
+
+    @StringForgery(regex = "[a-z]+\\.[a-z]+@[a-z]+\\.[a-z]{3}")
+    lateinit var fakeUserEmail: String
+    private var fakeUserAdditionalProperties: Map<String, Any?> = emptyMap()
+    private lateinit var stubFeature: StorageBackedFeature
+
+    @StringForgery(type = StringForgeryType.ALPHABETICAL)
+    lateinit var fakeFeatureName: String
+
+    private lateinit var fakeTrackingConsent: TrackingConsent
+
+    private var featureSdkCore: FeatureSdkCore? = null
```

Comment:
better to use `lateinit var` with non-nullable type. Will remove a lot of `.?` in the code.